src/server/domain/thirdpartyRequests/transactions/{ID}/authorizations.ts

```diff
@@ -0,0 +1,86 @@
+/*****
+ License
+ --------------
+ Copyright © 2020 Mojaloop Foundation
+ The Mojaloop files are made available by the Mojaloop Foundation under the
+ Apache License, Version 2.0 (the 'License') and you may not use these files
+ except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop
+ files are distributed onan 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ ANY KIND, either express or implied. See the License for the specific language
+ governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Raman Mangla <ramanmangla@google.com>
+ --------------
+ ******/
+
+import { Consent } from '../../../../../model/consent'
+import { Scope } from '../../../../../model/scope'
+
+/*
+ * Interface for incoming payload
+ */
+export interface AuthPayload {
+  consentId: string;
+  sourceAccountId: string;
+  status: string;
+  challenge: string;
+  value: string;
+}
+
+/*
+ * Domain function to validate against NULL fields
+ */
+export function hasNullFields (payload: AuthPayload): boolean {
+  for (const key in payload) {
+    if (payload[key as keyof AuthPayload] === null) {
+      return true
+    }
+  }
+
+  return false
+}
+
+/*
+ * Domain function to validate payload status
+ */
+export function hasCorrectStatus (payload: AuthPayload): boolean {
+  return payload.status === 'PENDING'
+}
+
+/*
+ * Domain function to check for existence of an active Consent key
+ */
+export function hasActiveConsentKey (consent: Consent): boolean {
```

Comment:
the name of the function doesn't match with implementation details
```suggestion
export function hasActiveCredentialWithPayload (consent: Consent): boolean {
```

Reply:
Are you sure this is better? The intent here is to check whether the `consent` in the DB has an active key credential. The active status is not really related to the payload.

Reply:
I'm confused what we mean by "key" here. Is this a key in a map? We don't talk about "keys" in the API, right? "Public key" is considered a "credential".

I think the code we have here checks... whether the consent's credential is active and has a payload. So @eoln 's description makes way more sense.